### PR TITLE
Android Studio 2.0 Upgrade (build tools 23.0.3 + gradle 2.10)

### PR DESCRIPTION
**Pull request by AndyScherzinger**
The new Android Studio 2.0 uses gradle 2.10 and also comes with a new build tools version.
This is the corresponding library PR for the android app PR https://github.com/owncloud/android/pull/1596

What has been done:

- Update Gradle to 2.10
- Update build tools to 23.0.3
- Updated projects to use Android plugin 2.0.0

### CODE REVIEW

**Comment by davivel** [comment]
And having a look here too.

**Comment by davivel** [comment]
Related: https://github.com/owncloud/android/pull/1596

File: .travis.yml [hunk 1]
```diff
@@ -13,6 +13,8 @@ branches:
 before_install:
 - echo no | android create avd --force -n test -t $ANDROID_TARGET --abi $ANDROID_ABI
   -c 20M
+- echo yes | android update sdk --filter extra-android-support --no-ui --force > /dev/null
```
Comment (davivel) [inline .travis.yml]: Why are these needed?

Comment (AndyScherzinger) [inline .travis.yml]: The build process does not update the android support lib artefacts in the local repo and thus upgrading to a support lib version newer than 20.x fails since it can't be found in a maven repo. So this android call updates the support libs so the dependency can be resolved. It is just cosmetics. This can be removed if the gradle file is reverted back to reference support lib v20.0.0

Comment (davivel) [inline .travis.yml]: Travis recommends doing these updates through the ```components``` section.

I'd give it a try to add ``` - extra-android-m2repository``` there, and maybe also ```- extra-android-support```

Comment (davivel) [inline .travis.yml]: Moreover, checking the project, seems there is no real dependency on the support library in this repo; I'll double check in command line; in that case, we should fully drop that dependency.

Comment (AndyScherzinger) [inline .travis.yml]: Okay, I'll check if we can get rid of it completely :)

Comment (AndyScherzinger) [inline .travis.yml]: Yeah! Removed! :) The support lib isn't used at all! Nice catch!

**Comment by AndyScherzinger** [comment]
Code review comments integrated.

**Comment by AndyScherzinger** [comment]
@davivel the PR is imho done since the travis build is working just fine. not sure if there are any tests/checks that need to be done for oC internal/non-pulic build processes. So it tagged it with Ready to test :)

**Comment by davivel** [comment]
Sorry, @AndyScherzinger, but the (main/last) committer of a PR cannot decide alone if contribution is OK from a technical standpoint.

I will resume my review today later.

**Comment by davivel** [comment]
This is OK. :+1:

Tested that works with Gradle in command line (via gradle wrapper in the project), works in Android Studio 2.0, and keeps working in Android Studio 1.5 too.

Thanks a lot, @AndyScherzinger

**Comment by AndyScherzinger** [comment]
As always you're welcome and I am happy to help! :) Sorry for prematurely tagging it ready for test! I misinterpreted the review comment as you being done and overlooked the missing ":+1: comment".
